gammapy/modeling/models/core.py

```diff
@@ -237,6 +242,58 @@ def reassign(self, datasets_names, new_datasets_names):
                         model.datasets_names[k] = new_dataset_name
         return model
 
+    def contribute(self, mask, psf=None):
```

Comment:
Again I would change the signature to `contributes(mask, margin="0.6 deg", inside=True)`. It's more flexible and easier to use. Again the connection to the `PSFKernel` could be done just with a little docs example. Internally to apply the margin I would rely on the `WcsNDMap.erode()`.

Reply:
For now I would prefer to not apply the margin using WcsNDMap.erode() so it match the behaviour of the MapEvaluator. Then I will check if the MapEvaluator can be adapted to use .contribute() and make an other PR if so.